rvm.el

```diff
@@ -351,14 +325,74 @@ function."
     (setq eshell-path-env (getenv "PATH"))
     (set current-binary-var new-binaries)))
 
+(defun rvm--find-rc-info (&optional path)
+  (let* ((cfiles (list rvm-configuration-file-name rvm-configuration-ruby-version-file-name rvm-configuration-gemfile-file-name))
+	 (path (or path (expand-file-name (or buffer-file-name ""))))
+	 (winner (rvm--locate-dominating-file
+		  path
+		  (lambda (dir)
+		    (when
+			(delq nil (mapcar
+				   (lambda (file)
+				     (file-exists-p (concat dir "/" file)))
+				   cfiles))
+		      t))))
+	 config-file-path)
+    (cond
+     ((not winner)
+      nil)
+     ((file-exists-p (setq config-file-path (concat winner rvm-configuration-file-name)))
+      (rvm--rvmrc-read-version config-file-path))
+     ((file-exists-p (setq config-file-path (concat winner rvm-configuration-ruby-version-file-name)))
+      (list (chomp (rvm--get-string-from-file config-file-path))
+	    (if (file-exists-p (setq config-file-path (concat winner rvm-configuration-ruby-gemset-file-name)))
+		(chomp (rvm--get-string-from-file config-file-path))
+	      rvm--gemset-default)))
+     ((file-exists-p (setq config-file-path (concat winner rvm-configuration-gemfile-file-name)))
+      (rvm--gemfile-read-version config-file-path)))))
+
+(defun rvm--locate-dominating-file (file name)
+  "Look up the directory hierarchy from FILE for a directory containing NAME.
+Stop at the first parent directory containing a file NAME,
+and return the directory.  Return nil if not found.
+Instead of a string, NAME can also be a predicate taking one argument
+\(a directory) and returning a non-nil value if that directory is the one for
+which we're looking."
+  ;; Heisted from files.el:24.3.50.2.
+  ;; Represent /home/luser/foo as ~/foo so that we don't try to look for
+  ;; `name' in /home or in /.
+  (setq file (abbreviate-file-name file))
+  (let ((root nil)
+        ;; `user' is not initialized outside the loop because
+        ;; `file' may not exist, so we may have to walk up part of the
+        ;; hierarchy before we find the "initial UID".  Note: currently unused
+        ;; (user nil)
+        try)
+    (while (not (or root
+                    (null file)
+                    ;; FIXME: Disabled this heuristic because it is sometimes
+                    ;; inappropriate.
```

Comment:
Is this the only reason to bring this in from `files.el`?

Under what circumstances is it inappropriate?